R/convert_creat_unit.R

```diff
@@ -11,27 +11,8 @@ convert_creat_unit <- function(value,
                                unit_out = valid_units("scr")) {
   unit_in <- tolower(unit_in)
   unit_out <- tolower(unit_out)
-  if (length(unit_in) != length(value) && length(unit_in) != 1) {
-    stop("length of unit_in must be either 1 or the same as values")
-  }
   unit_in <- match.arg(unit_in, several.ok = TRUE)
   unit_out <- match.arg(unit_out)
 
-  conv <- c(
-    `mg/dl`      = 1,
-    mg_dl        = 1,
-    `micromol/l` = 1 / 88.42,
-    micromol_l   = 1 / 88.42,
-    micromol     = 1 / 88.42,
-    mmol         = 1 / 88.42,
-    `mumol/l`    = 1 / 88.42,
-    `mumol_l`    = 1 / 88.42,
-    `umol/l`     = 1 / 88.42,
-    `umol_l`     = 1 / 88.42
-  )
-
-  list(
-    value = value * unname(conv[unit_in]) / unname(conv[unit_out]),
-    unit = unit_out
-  )
+  convert_conc_unit(value, unit_in, unit_out, 1/88.42 * 10000)
```

Comment:
is it maybe more legible to write this as 113.12? 
```suggestion
  convert_conc_unit(value, unit_in, unit_out, 113.12) # MW = 1/88.42 * 10000
```

Reply:
I agree it's more legible, however I just tried it and discovered that the value is not the same and some of the test values will change. 1/88.42 * 10000 is 113.0966 not 113.12. This might affect things in other codebases as well. We've had 1/88.42 since this function was written. But, when you google "creatinine molecular weight" you get 113.12 😬  not sure what's the best way to proceed here, maybe I should change it and try running some downstream tests to see if it breaks things there?

Reply:
hmmm strange rounding issue to encounter. yeah maybe it makes sense to see if it has a high impact, otherwise I guess we can leave it as 1/88.42 or 113.1.